service/test/BatchServerTest.cpp

```diff
@@ -359,6 +461,263 @@ TEST_F(BatchServerTest, run_batch_write_empty)
     m_batch_server_signals->run_batch();
 }
 
+/**
+ * @test Provides coverage for the case when BatchStatus::M_MESSAGE_TERMINATE is read message
+ *       by the server in BatchServerImp::event_loop()
+ */
+TEST_F(BatchServerTest, receive_message_terminate)
+{
+    InSequence sequence;
+
+    int idx = 0;
+    std::vector<double> result = {240.042, 250.052};
+
+    for (const auto &request : m_signal_config) {
+        EXPECT_CALL(*m_pio_ptr, push_signal(request.name, request.domain,
+                                            request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+    }
+
+    for (const auto &request : m_control_config) {
+        EXPECT_CALL(*m_pio_ptr, push_control(request.name, request.domain,
+                                             request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+    }
+
+    EXPECT_CALL(*m_batch_status, receive_message())
+        .WillOnce(Return(BatchStatus::M_MESSAGE_TERMINATE))
+        .RetiresOnSaturation();
+
+    m_batch_server->run_batch();
+}
+
+/**
+ * @test Provides coverage for the case when not a valid message is read
+ *       by the server in BatchServerImp::event_loop()
+ */
+TEST_F(BatchServerTest, receive_message_default)
+{
+    InSequence sequence;
+
+    int idx = 0;
+    std::vector<double> result = {240.042, 250.052};
+
+    for (const auto &request : m_signal_config) {
+        EXPECT_CALL(*m_pio_ptr, push_signal(request.name, request.domain,
+                                            request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+    }
+
+    for (const auto &request : m_control_config) {
```

Comment:
```suggestion
    idx = 0;
    for (const auto &request : m_control_config) {
```

When pushing signals and controls, make sure the index is reset in between.  It doesn't matter for this test because you are not verifying the indices, but it will lead to confusing behavior if this code is used in other tests that do utilizes the indices.